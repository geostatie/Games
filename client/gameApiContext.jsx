import React from "react";
import { fetchJSON } from "./fetchJSON.jsx";

export const GameApiContext = React.createContext({
  async register(user){
    fetch("/api/games/register", {
        method: "post",
        body: JSON.stringify(user),
        headers: {
          "content-type": "application/json",
        }
    });
  },
  async listGames(title){
        return await fetchJSON(`/api/games?titleSearch=${title}`);
    },
    async createGame(game){
      fetch("/api/games", {
          method: "post",
          body: JSON.stringify(game),
          headers: {
            "content-type": "application/json",
          },
      });
    }
});