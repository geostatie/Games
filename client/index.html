<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Back to the games we go</title>
</head>
<body>
<div id="app">
</div>
</body>
<script src="index.jsx" type="module" />
</html>