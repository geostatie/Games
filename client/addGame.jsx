import * as React from "react";
import {useNavigate} from "react-router-dom";
import { useContext, useState } from "react";
import { GameApiContext } from "./gameApiContext.jsx";


function FormInput({value, label, onChangeValue}) {
    return(
      <div>
          <label>
              <strong>{label} </strong>
              <input value={value} onChange={(e) => onChangeValue(e.target.value)} />
          </label>
      </div>
    );
}

export function AddGame(){
    const { createGame } = useContext(GameApiContext);
    const navigate = useNavigate();

    const [title, setTitle] = useState("");
    const [yearInput, setYearInput] = useState("");
    const [plot, setPlot] = useState("");


    async function handleSubmit(e){
        e.preventDefault();
        const year = parseInt(yearInput);
        createGame({ title, year, plot });
        navigate("/");
    }

    return(
        <form onSubmit={handleSubmit}>
            <h1> Submit new game </h1>
            <FormInput label="Title:" value={title} onChangeValue={setTitle}/>
            <FormInput label="Year:" value={yearInput} onChangeValue={setYearInput}/>
            <FormInput label="Plot:" value={plot} onChangeValue={setPlot}/>
            <button> Submit </button>
        </form>
    );
}

