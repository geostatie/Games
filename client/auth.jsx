import React, { useContext, useState } from "react";
import {GameApiContext} from "./gameApiContext.jsx";
import {useLoader} from "./useLoader.jsx";

export function AuthPage() {
 
    return (
        <div>
            
        </div>
    );
}