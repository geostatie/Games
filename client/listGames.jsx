import React, { useContext, useState } from "react";
import {GameApiContext} from "./gameApiContext.jsx";
import {useLoader} from "./useLoader.jsx";

function GameCard( { game: { title, plot, year } }) {
    return(
        <>
            <h3> {title} - {year} </h3>
            <div> {plot} </div>
        </>
    );
}


export function ListGames() {
    const { listGames } = useContext(GameApiContext);
    const [titleSearch, setTitleSearch] = useState("in");
    const [titleInput, setTitleInput] = useState("");
    const { loading, error, data } = useLoader(
        async () => await listGames( titleSearch ),
        [titleSearch]
    );

    function handleSubmit(e){
        e.preventDefault();
        setTitleSearch(titleInput);
    }

    if(loading) {
        return <div className="loading-indicator"> Still Loading... </div>;
    }

    if(error) {
        return (
            <div>
                <h1> Error </h1>
                <div className="error-message"> {error.toString()} </div>
            </div>
        );
    }

    return (
        <div>
            <h1> Games to come back to: </h1>
            <div>
                <form onSubmit={handleSubmit}>
                    <label> Search by title:
                    <input
                        id="search-spot"
                        value={titleInput}
                        onChange={(e) => setTitleInput(e.target.value)}
                    />
                    <button> Search </button>
                    </label>
                </form>
            </div>
            {data.map((game) => (
                <GameCard key={game.title} game={game} />
            ))}
        </div>
    );
}