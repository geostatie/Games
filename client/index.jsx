import * as React from "react";
import { createRoot } from "react-dom/client";

import { Application } from "./Games.jsx";

const element = document.getElementById("app");
const root = createRoot(element);

root.render(<Application />);