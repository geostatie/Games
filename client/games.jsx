import * as React from "react";
import {BrowserRouter, Link, Route, Routes } from "react-router-dom";

import { AddGame } from "./addGame.jsx";
import { ListGames } from "./listGames.jsx";

export function FrontPage() {
    return(
        <div>
            <h1> Back to the games </h1>
            <ul>
                <li>
                    <Link to={"/games/list"}> List Games </Link>
                </li>
                <li>
                    <Link to={"/games/new"}> Add new Game </Link>
                </li>
            </ul>
        </div>
    );
}

export function Games(){
    return(
        <Routes>
            <Route path={"/list"} element={<ListGames />} />
            <Route path={"/new"} element={<AddGame />} />
        </Routes>
    );
}

export function Application() {
    return(
        <BrowserRouter>
            <Routes>
                <Route path={"/"} element={<FrontPage />} />
                <Route path={"/games/*"} element={<Games />}/>
            </Routes>
        </BrowserRouter>
    );
}