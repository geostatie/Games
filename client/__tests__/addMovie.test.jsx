import * as React from "react";
import {createRoot} from "react-dom/client";
import { act, Simulate } from "react-dom/test-utils";
import { AddGame } from "../addGame.jsx";
import { MemoryRouter } from "react-router-dom";
import { GameApiContext } from "../gameApiContext.jsx";

describe("add game tests", () => {
    it("shows form", async () => {
       const element = document.createElement("div");
       const root = createRoot(element);

       await act(async () => {
           root.render(
            <MemoryRouter>
                <AddGame />
            </MemoryRouter>
           );
       });
       expect(element.innerHTML).toMatchSnapshot();

       const inputLabels = Array.from(
           element.querySelectorAll("form label strong")
       ).map((label) => label.innerHTML);
       expect(inputLabels).toEqual(["Title: ", "Year: ", "Plot: "]);
    });

    it("submits form", async () => {
        const createGame = jest.fn();

       const element = document.createElement("div");
       const root = createRoot(element);

       await act( async () =>
            root.render(
            <MemoryRouter>
                <GameApiContext.Provider value={{ createGame }}>
                    <AddGame />
                </GameApiContext.Provider>
            </MemoryRouter>
            )
       );

       act ( () =>
       Simulate.change(element.querySelector("form div:nth-of-type(1) input"), {
          target: { value: "Game Title" },
       }));

       act( () =>
        Simulate.change(element.querySelector("form div:nth-of-type(2) input"), {
            target: { value: "2022" },
        }));

        act ( () =>
            Simulate.submit(element.querySelector("form"))
        );

       expect(createGame).toBeCalledWith({
          title: "Game Title",
          year: 2022,
          plot: "",
       });

    });
});