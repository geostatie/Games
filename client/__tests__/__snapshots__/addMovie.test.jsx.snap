// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add game tests shows form 1`] = `"<form><h1> Submit new game </h1><div><label><strong>Title: </strong><input value=""></label></div><div><label><strong>Year: </strong><input value=""></label></div><div><label><strong>Plot: </strong><input value=""></label></div><button> Submit </button></form>"`;
